main/src/ca/uwaterloo/flix/language/phase/Namer.scala

```diff
@@ -89,12 +88,34 @@ object Namer extends Phase[WeededAst.Program, NamedAst.Root] {
     case decl@WeededAst.Declaration.Class(doc, mod, ident, tparam, sigs, loc) =>
       // Check if the class already exists.
       val classes = prog0.classes.getOrElse(ns0, Map.empty)
+      val sigNs = Name.extendNName(ns0, ident)
+      val defsAndSigs0 = prog0.defsAndSigs.getOrElse(sigNs, Map.empty)
       classes.get(ident.name) match {
         case None =>
           // Case 1: The class does not already exist. Update it.
-          visitClass(decl, uenv0, Map.empty, ns0) map {
-            case clazz@NamedAst.Class(_, _, _, _, sigs0, _) =>
-              prog0.copy(classes = prog0.classes + (ns0 -> (classes + (ident.name -> clazz))))
+          visitClass(decl, uenv0, Map.empty, ns0) flatMap {
+            case clazz@NamedAst.Class(_, _, _, _, sigs, _) =>
+              // check whether each sig is already in the root
```

Comment:
I don't understand this check. What is meant by "root"?

Reply:
Leftover from a previous iteration; should be something like "add each signature to the namespace"